### DISCUSSION TITLE
How to jump to `var` and `struct` definitions using `clangd` via `gd/gD` in project?

### DISCUSSION THREAD

**Discussion by longintlong** [ayamir/nvimdots#836] [Q&A] [answered]
在一个项目内部，很多变量结构体都提示unknown type，或者gD 跳转不过去，请问是需要配置什么东西吗

**Comment by Jint-lzxy** [comment]
Can u help fill in the [issue form](https://github.com/ayamir/nvimdots/issues/new?assignees=&labels=bug%2Clsp&projects=&template=lsp_issue_report.yml)? This discussion provides too little information, and we can't _infer_ the problem from a simple description. Instead, we need a specific environment (e.g., which server?), steps to reproduce this issue, and relevant logs to know exactly what the problem is. Such a bald description is not helpful at all.

**Reply by longintlong** [reply]
抱歉
我的环境是fedora 36， nvim master，mason install clang ，gcc version 为 12，下面这张图是复现的问题，discard_sector_alignment 这个变量，我想查看这个变量在结构体中的定义，所以我使用 gd，但是右上角lsp 提示 def 是 nil
<img width="1850" alt="image" src="https://github.com/ayamir/nvimdots/assets/39492651/9b96ed17-e380-43d5-b097-18baa8a0b339">

**Reply by Jint-lzxy** [reply]
Which branch are u using? ur local branch seems to be _a lot_ behind the remote _(we don't have a 'master' branch)_.

Can't repro this:
<img width="320" alt="demo" src="https://github.com/ayamir/nvimdots/assets/50296129/e6b1c98f-83c9-495d-98ac-943cc61fd02e">

A common cause of this issue is u didn't provide a JSON Compilation Database. Please refer to [`clangd`'s documentation](https://clangd.llvm.org/design/compile-commands) and the pinned issues for more information.

***
btw, u don't seem to have the required font installed correctly. u may download ur desired font [here](https://www.nerdfonts.com/font-downloads) or via [Homebrew Cask Fonts](https://github.com/ryanoasis/nerd-fonts#option-4-homebrew-fonts), and then select the font ending with `Nerd Font Mono` _(e.g., JetBrainsMono Nerd Font Mono)_ in your terminal's custom profile.

**Comment by longintlong** [comment]
branch 0.8，您的这个例子是在同一个文件，我又做了一个简单的测试如temp.cpp 和 temp.h，.h文件定义了struct A，在cpp文件中 gd 确实是可以找到的。
我的实际场景是在一个project中，您的意思是需要在我的project中添加json compile database吗？

----
感谢提醒字体，这个应该不会影响我的跳转，我之后会下载，现在刚开始使用vim，有许多快捷键不清楚如何使用。
再次感谢您的项目

**Reply by charliie-dev** [reply]
0.8 is too old, we cant give any promises if older branch will also works fine on every functionality.

**Reply by longintlong** [reply]
好的，那我尝试拉取main 分支的最新代码

**Reply by charliie-dev** [reply]
try 0.9 branch

**Reply by longintlong** [reply]
我尝试了main分支，出现了同样的错误。安装脚本中 clone 的是 main 分支或者 0.8分支

**Reply by Jint-lzxy** [reply, marked as answer]
> 您的意思是需要在我的project中添加json compile database吗？

ys. This is the only way `clangd` can understand complex code trees. VSCode doesn't require such configuration b/c it has already generated this file for you (usually via [`rizsotto/bear`](https://github.com/rizsotto/Bear)).

Make sure to read [`clangd`'s documentation](https://clangd.llvm.org/design/compile-commands) before u start :)

**Reply by longintlong** [reply]
感谢
